source/dissocPath.js

```diff
@@ -27,18 +46,20 @@ import update from './update';
  *      R.dissocPath(['a', 'b', 'c'], {a: {b: {c: 42}}}); //=> {a: {b: {}}}
  */
 var dissocPath = _curry2(function dissocPath(path, obj) {
+  if (obj == null) {
+    return obj;
+  }
+
   switch (path.length) {
     case 0:
       return obj;
     case 1:
-      return _isInteger(path[0]) && _isArray(obj) ? remove(path[0], 1, obj) : dissoc(path[0], obj);
+      return _dissoc(path[0], obj);
     default:
       var head = path[0];
       var tail = Array.prototype.slice.call(path, 1);
       if (obj[head] == null) {
-        return obj;
-      } else if (_isInteger(head) && _isArray(obj)) {
-        return update(head, dissocPath(tail, obj[head]), obj);
+        return _shallowCloneObject(head, obj);
```

Comment:
I cloned `obj` when path can not deep in

But I can not clone `obj` when path is empty, because I have no way to judge how to clone it (Ramda clone `obj` as an array is according to `_isInteger(prop) && _isArray(obj)`, but I can not get `prop` in that clause)